pdks/Nonuniform_mock_pdk/primitive.py

```diff
@@ -0,0 +1,207 @@
+import os
+import json
+import math
+import pathlib
+import importlib
+from itertools import cycle, islice, chain
+from align.cell_fabric import transformation
+from canvas import NonuniformCanvas
+from align.schema.transistor import Transistor, TransistorArray
+from gen_transistor import mos
+
+
+class MOSGenerator(NonuniformCanvas):
```

Comment:
MOSGenerator needs to become ArrayGenerator (I see you already created a mos wrapper with this in mind). Would defer this to a separate PR.

Reply:
Good point. I intentionally kept compatibility with the current flow. This class and the interface between could be further simplified when we adopt the new schema.